src/plugins/embeddable/public/plugin.test.ts

```diff
@@ -33,3 +35,26 @@ test('cannot register embeddable factory with the same ID', async () => {
     'Embeddable factory [embeddableFactoryId = ID] already registered in Embeddables API.'
   );
 });
+
+test('can set custom embeddable factory provider', async () => {
+  const coreSetup = coreMock.createSetup();
+  const coreStart = coreMock.createStart();
+  const { setup, doStart } = testPlugin(coreSetup, coreStart);
+
+  const customProvider: EmbeddableFactoryProvider = def => ({
+    ...defaultEmbeddableFactoryProvider(def),
+    getDisplayName: () => 'Intercepted!',
+  });
+
+  setup.setCustomEmbeddableFactoryProvider(customProvider);
+  setup.registerEmbeddableFactory('test', {
+    type: 'test',
+    create: () => Promise.resolve(undefined),
+    getDisplayName: () => 'Test',
+    isEditable: () => Promise.resolve(true),
+  });
+
+  const start = doStart();
+  const factory = start.getEmbeddableFactory('test');
+  expect(factory!.getDisplayName()).toEqual('Intercepted!');
+});
```

Comment:
Maybe we can add a use case test here how `setCustomEmbeddableFactoryProvider` would be used with dynamic actions&mdash;hook into "start, "stop" life-cycles and "input" changes of each embeddable.